Add support for btrfs subvolumes, metadata profiles and compression

ondrejbudai opened the pull request:
This PR finishes support for btrfs in osbuild. This means:

- support for btrfs subvolumes, osbuild can now both create them and mount them
- support for setting the metadata profile for btrfs. We should always set this to `dup`, but it's nice to have it exposed in the stage options.
- support for mounting with compression
- an example of a manifest

thozza commented:
WRT testing the functionality... From what I saw in the repo, we AFAICT do not test stages that are used to create filesystem explicitly. Some information related to testing osbuild stages is listed on https://www.osbuild.org/guides/developer-guide/osbuild.html#stage-development, but I don't think it will be very useful in this case.

You may need to implement a special test case, such as those implemented below https://github.com/osbuild/osbuild/blob/main/test/run/test_stages.py#L245

Conan-Kudo commented:
Can we also set compression like we do with `appliance-creator`/`lorax`/`kiwi` based images?

Cf. https://fedoraproject.org/wiki/Changes/BtrfsTransparentCompression

ondrejbudai commented:
I added a test that thozza suggested, rebased on top of main and polished it a bit. This is now ready to be reviewed and merged.

@Conan-Kudo Yes, compression is fully supported (both in fstab and during build time).

thozza reviewed (commented):
Nice work!

Thanks for the test. I found just one small issue. Besides this, you'll need to update the `osbuild-ci` container to contain `btrfs` binary -> https://github.com/osbuild/containers/blob/1884fa280b4faae0f5bd980b3bf9fc18a53c1098/docker-bake.hcl#L227

ondrejbudai commented:
Fixed missing `additionalProperties: false` in the subvol stage and added a marker to the new test that disables it when `--unsupported-fs btrfs` is passed to pytest.

The only blocker is now https://github.com/osbuild/containers/pull/56 I think.

thozza reviewed (approved):
LGTM

ondrejbudai commented:
Updated the reference to the testing container image to the one with `btrfs-progs` installed.

ondrejbudai commented:
Thanks for starting this, @gicmo. ♥️
